### ISSUE TITLE
shell crashes when enter is pressed

### ISSUE THREAD

**Issue by rmstoi** [zephyrproject-rtos/zephyr#6322]
There seems to be a regression in shell that causes it to crash after enter is pressed. For example if I start echo server in QEMU and press enter it crashes with message:

```
***** CPU Page Fault (error code 0x00000002)
Supervisor thread wrote address 0x00000005
PDE: 0x025 Present, Read-only, User, Execute Enabled
PTE: 0x00 Non-present, Read-only, Supervisor, Execute Enabled
Current thread ID = 0x00404540
Faulting segment:address = 0x0008:0x00003c98
eax: 0x0041f00d, ebx: 0x00000000, ecx: 0x00000000, edx: 0x0041f900
esi: 0x00000000, edi: 0x0041b7e3, ebp: 0x0041b7f0, esp: 0x0041b7d8
eflags: 0x246
Fatal fault in ISR! Spinning...
Terminate emulator due to fatal kernel error

```

There is a PR #6313 with a patch.

It seems regression is probably caused by commit:

[14735116d1fe24713aa444e4e38e313b7f117c89](https://github.com/zephyrproject-rtos/zephyr/commit/14735116d1fe24713aa444e4e38e313b7f117c89#diff-d3cf833345c74a1b1e4ae843036ff7b8L421)

Basically line where continue is deleted

```
-               /* Handle special control characters */
                if (!isprint(byte)) {
+                       /* Handle special control characters */
                        switch (byte) {
                        case DEL:
                                if (cur > 0) {
@@ -417,8 +541,6 @@ void uart_console_isr(struct device *unused)
                        default:
                                break;
                        }
-
-                       continue;
                }

```
